### ISSUE TITLE
Getting "OOPS" Page error when powermail is included on root page

### ISSUE THREAD

**Issue by ColonelMillerSG21** [einpraegsam/powermail#18]
I should mention that i'm using RealUrl with t3 v.6.1.
If i add the powermail extension to my rootpage, realurl seems to cut of the page id which leads to the page not loading when i submit my powermail form: 
![image](https://cloud.githubusercontent.com/assets/11478472/21415901/a55ba590-c80e-11e6-8140-59bf8ae3b916.png)
Is there any way to solve this?

**Comment by einpraegsam** [comment]
What is the exactly issue? You push submit and the form is still shown or is there an exception?

**Comment by ColonelMillerSG21** [comment]
I push submit and get this: 
![image](https://cloud.githubusercontent.com/assets/11478472/21436910/2261070c-c881-11e6-8016-a52560f39480.png)

**Comment by ColonelMillerSG21** [comment]
It works on any other page except the rootpage where i get this.

**Comment by einpraegsam** [comment]
ok, and what exactly is the error? Can you please activate errors or have a look into the apache error log.

**Comment by ColonelMillerSG21** [comment]
Uncaught TYPO3 Exception
#1298012500: Required argument "form" is not set. (More information)
TYPO3\CMS\Extbase\Mvc\Controller\Exception\RequiredArgumentMissingException thrown in file
/typo3/sysext/extbase/Classes/Mvc/Controller/AbstractController.php in line 440.

14 TYPO3\CMS\Extbase\Mvc\Controller\AbstractController::mapRequestArgumentsToControllerArguments()

/typo3/sysext/extbase/Classes/Mvc/Controller/ActionController.php:
00177:    call_user_func(array($this, $actionInitializationMethodName));
00178:   }
00179:   $this->mapRequestArgumentsToControllerArguments();
00180:   $this->checkRequestHash();
00181:   $this->controllerContext = $this->buildControllerContext();

13 TYPO3\CMS\Extbase\Mvc\Controller\ActionController::processRequest(TYPO3\CMS\Extbase\Mvc\Web\Request, TYPO3\CMS\Extbase\Mvc\Web\Response)

/typo3/sysext/extbase/Classes/Mvc/Dispatcher.php:
00098:    $controller = $this->resolveController($request);
00099:    try {
00100:     $controller->processRequest($request, $response);
00101:    } catch (\TYPO3\CMS\Extbase\Mvc\Exception\StopActionException $ignoredException) {
00102:    }

12 TYPO3\CMS\Extbase\Mvc\Dispatcher::dispatch(TYPO3\CMS\Extbase\Mvc\Web\Request, TYPO3\CMS\Extbase\Mvc\Web\Response)

/typo3/sysext/extbase/Classes/Mvc/Web/FrontendRequestHandler.php:
00082:   /** @var $response \TYPO3\CMS\Extbase\Mvc\ResponseInterface */
00083:   $response = $this->objectManager->get('TYPO3\\CMS\\Extbase\\Mvc\\Web\\Response');
00084:   $this->dispatcher->dispatch($request, $response);
00085:   return $response;
00086:  }

11 TYPO3\CMS\Extbase\Mvc\Web\FrontendRequestHandler::handleRequest()
/typo3/sysext/extbase/Classes/Core/Bootstrap.php:
00206:   $requestHandler = $requestHandlerResolver->resolveRequestHandler();
00207: 
00208:   $response = $requestHandler->handleRequest();
00209:   // If response is NULL after handling the request we need to stop
00210:   // This happens for instance, when a USER object was converted to a USER_INT

10 TYPO3\CMS\Extbase\Core\Bootstrap::handleRequest()

/typo3/sysext/extbase/Classes/Core/Bootstrap.php:
00195:  public function run($content, $configuration) {
00196:   $this->initialize($configuration);
00197:   return $this->handleRequest();
00198:  }
00199: 

9 TYPO3\CMS\Extbase\Core\Bootstrap::run("", array)
8 call_user_func_array(array, array)

/typo3/sysext/frontend/Classes/ContentObject/ContentObjectRenderer.php:
06439:       $content,
06440:       $conf
06441:      ));
06442:     } else {
06443:      $GLOBALS['TT']->setTSlogMessage('Method "' . $parts[1] . '" did not exist in class "' . $parts[0] . '"', 3);

7 TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer::callUserFunction("TYPO3\CMS\Extbase\Core\Bootstrap->run", array, "")

/typo3/sysext/frontend/Classes/ContentObject/UserContentObject.php:
00053:   }
00054:   $this->cObj->includeLibs($conf);
00055:   $tempContent = $this->cObj->callUserFunction($conf['userFunc'], $conf, '');
00056:   if ($this->cObj->doConvertToUserIntObject) {
00057:    $this->cObj->doConvertToUserIntObject = FALSE;

6 TYPO3\CMS\Frontend\ContentObject\UserContentObject::render(array)

/typo3/sysext/frontend/Classes/ContentObject/ContentObjectRenderer.php:
00902:    return $this->getContentObject('USER_INT')->render($conf);
00903:   } else {
00904:    return $this->getContentObject('USER')->render($conf);
00905:   }
00906:  }

5 TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer::USER(array)

/typo3/sysext/frontend/Classes/Controller/TypoScriptFrontendController.php:
03478:       break;
03479:      case 'FUNC':
03480:       $incContent = $INTiS_cObj->USER($INTiS_config[$INTiS_key]['conf']);
03481:       break;
03482:      case 'POSTUSERFUNC':

4 TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController::INTincScript_process(array)

/typo3/sysext/frontend/Classes/Controller/TypoScriptFrontendController.php:
03426:    $INTiS_config = $this->config['INTincScript'];
03427:    $this->INTincScript_includeLibs($INTiS_config);
03428:    $this->INTincScript_process($INTiS_config);
03429:    // Check if there were new items added to INTincScript during the previous execution:
03430:    $INTiS_config = array_diff_assoc($this->config['INTincScript'], $INTiS_config);

3 TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController::recursivelyReplaceIntPlaceholdersInContent()

/typo3/sysext/frontend/Classes/Controller/TypoScriptFrontendController.php:
03404:    $this->setPageRenderer(unserialize($this->config['INTincScript_ext']['pageRenderer']));
03405:   }
03406:   $this->recursivelyReplaceIntPlaceholdersInContent();
03407:   $GLOBALS['TT']->push('Substitute header section');
03408:   $this->INTincScript_loadJSCode();

2 TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController::INTincScript()

/typo3/sysext/cms/tslib/index_ts.php:
00202: if ($TSFE->isINTincScript()) {
00203:  $TT->push('Non-cached objects', '');
00204:  $TSFE->INTincScript();
00205:  $TT->pull();
00206: }

1 require("/typo3/sysext/cms/tslib/index_ts.php")

/index.php:
00039:  ->redirectToInstallToolIfLocalConfigurationFileDoesNotExist();
00040: 
00041: require(PATH_tslib . 'index_ts.php');
00042: ?>

**Comment by einpraegsam** [comment]
"Required argument "form" is not set." That means that there are POST parameters missing. Did you check if there is a redirect (maybe added via .htaccess) with your browser console?
Or another way: Please check the source code of you home page, search for the form tag, search for the action parameter. There must be an URI. Please copy it and try to call in in your browser. Is there a redirect?

**Comment by einpraegsam** [comment]
Any feedback for me? If not, I will close this issue.

**Comment by ColonelMillerSG21** [comment]
The URI in my sourcecode looks like this:
?tx_powermail_pi1%5Baction%5D=create&tx_powermail_pi1%5Bcontroller%5D=Forms&cHash=03ca25d0581a392398d6f149b360e6e1

If i manually call it i get the same error. Looks like realurl is getting rid of the index.php?id=3 part.

**Comment by einpraegsam** [comment]
Yes, of course. If you submit, a mix from GET and POST is used in powermail. If you call it directly in your browser without the needed params (that are normally sent via POST), an error will come up.
Is it an official page where I can look at it?
To your realurl issue: Can you turn off realurl for testing?

**Comment by ColonelMillerSG21** [comment]
It works with realurl turned off but i need realurl. This is the url: adasmarket.com
It will open a lightbox when you first land there with powermail inside. 
It works perfectly fine on all other pages like the rewards page for example if you click on subscribe you can test it.

**Comment by einpraegsam** [comment]
Thx for the URI.

The first thing that I saw is:
The form action is "?tx_powermail_pi1%5Baction%5D=create&tx_powermail_pi1%5Bcontroller%5D=Forms&cHash=03ca25d0581a392398d6f149b360e6e1".
When I open the page http://www.adasmarket.com/ I can see a different baseUrl https://adasmarket.com/
That could be one of the issues.

A simple test let me submit the form. When I change the action with the browser console to "http://www.adasmarket.com/index.php?tx_powermail_pi1%5Baction%5D=create&tx_powermail_pi1%5Bcontroller%5D=Forms&cHash=03ca25d0581a392398d6f149b360e6e1" I can send the form without problems.

If I had to resolve this issue, I would let powermail create an absolute URI for the form action. 
In addition you should really upgrade your TYPO3 6.1 and your old powermail because of security reasons.

BTW: There is no more support for powermail that is older than version 3.0 and I think you're using version 1.x or 2.0 or so.

Long story short: This is not a powermail bug. So I will close the issue.

**Comment by ColonelMillerSG21** [comment]
Is there any documentation on how to set an absolute URI for the form action in powermail? Moving to 8.5 very soon.

**Comment by ColonelMillerSG21** [comment]
Fixed it thank you. It was the https setting you mentioned.
